pkg/s3wrapper/client.go

```diff
@@ -30,9 +30,14 @@ import (
 
 const awsEndpointSuffix = ".amazonaws.com"
 
-// TODO: Can we use RHCOS_VERSION or BASE_OS_IMAGE in the Makefile to set the value for
-// BaseObjectName?
-const BaseObjectName = "livecd-46.82.202009222340-0.iso"
+// TODO: Eventually we should make the base image piece managed out-of-band.
+// Here, we are providing a fallback in the case an image isn't provided.
+const RHCOSBaseURL = "https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.6/4.6.1/rhcos-4.6.1-x86_64-live.x86_64.iso"
+const RHCOSBaseObjectName = "rhcos-46.82.202010091720-0.iso"
+
+// We will need to modify this based on whatever is provided to the
+// assisted-service at runtime.
+var BaseObjectName string
```

Comment:
Can this be provided as a part of `NewS3Client` rather than a global? Maybe in the config somewhere?

Reply:
It can be. The reason I had gone with this approach was to avoid updating all the calls to `s3wrapper.BaseObjectName` to `s3wrapper.BaseObjectName()`.

It made me a little uneasy to make it a global like this, let me know if I should make it private and use accessors.

Reply:
I couldn't find an intelligent way to do this that didn't make integrating with https://github.com/openshift/assisted-service/pull/755 more painful for @avishayt so I'm skipping it and will circle back after #755 is merged.